x-pack/plugins/ml/public/application/jobs/new_job/job_from_pattern_analysis/utils.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DataViewField, DataView } from '@kbn/data-views-plugin/common';
+import type { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import type { SharePluginStart } from '@kbn/share-plugin/public';
+import type { TimeRange } from '@kbn/es-query';
+import { ML_APP_LOCATOR } from '../../../../../common/constants/locator';
+import { ML_PAGES } from '../../../../locator';
+import type { CategorizationType } from './quick_create_job';
+
+export async function redirectToADJobWizards(
+  categorizationType: CategorizationType,
+  dataView: DataView,
+  field: DataViewField,
+  partitionField: DataViewField | null,
+  stopOnWarn: boolean,
+  query: QueryDslQueryContainer,
+  timeRange: TimeRange,
+  share: SharePluginStart
+) {
+  const locator = share.url.locators.get(ML_APP_LOCATOR);
+
+  const url = await locator?.getUrl({
```

Comment:
Does there need to be a contingency for when locator is undefined? So the url doesn't end up as undefined?

Reply:
I think it's safe to assume that `locator` will alway be defined as we're in the ML app. We do this in other places.
I've updated the code to assert that `locator` is defined.
Updated in https://github.com/elastic/kibana/commit/d32017ded1a12c1aa0595d5f5d79dde82080995a